public/r2/v1/swagger/swagger.json

```diff
@@ -877,6 +943,66 @@
                 }
             }
         },
+        "RuleSetChange": {
+            "type": "object",
+            "properties": {
+                "operation": {
+                    "type": "string",
+                    "enum": ["add", "remove", "change"],
+                    "description": "type of operation on rule set"
+                },
+                "MappingRuleChanges": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/MappingRuleChange"
+                    },
+                    "description": "list of mapping rule diffs"
+                },
+                "RollupRuleChanges": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/RollupRuleChange"
+                    },
+                    "description": "list of rollup rule diffs"
+                }
+            }
+        },
+        "MappingRuleChange": {
+            "type": "object",
+            "properties": {
+                "operation": {
+                    "type": "string",
+                    "enum": ["add", "remove", "change"],
+                    "description": "type of operation on rule"
+                },
+                "from": {
+                    "$ref": "#/definitions/MappingRuleData",
+                    "description": "old version of rule"
+                },
+                "to": {
+                    "$ref": "#/definitions/MappingRuleData",
+                    "description": "new version of rule"
```

Comment:
Perhaps better to say "new rule data"?

Also this can be null for "remove" operations I assume?